src/strategies/tomyumswap/index.ts

```diff
@@ -0,0 +1,103 @@
+import fetch from 'cross-fetch';
+import { subgraphRequest } from '../../utils';
+
+export const author = 'tomyumswap';
+export const version = '0.0.1';
+
+type VotingResponse = {
+  verificationHash: string;
+  block: number;
+  tomYumBalance: string;
+  tomYumVaultBalance: string;
+  tomYumPoolBalance: string;
+  tomYumBnbLpBalance: string;
+  poolsBalance: string;
+  total: string;
+};
+
+const MINIMUM_VOTING_POWER = 0.01;
+const SMART_CHEF_URL = 'https://api.thegraph.com/subgraphs/name/tomyumswap/smartchef';
+const VOTING_API_URL = 'http://voting-api.tomyumswap.com/api/power';
+
+/**
+ * Fetches voting power of one address
+ */
+const fetchVotingPower = async (
+  address: string,
+  block: number,
+  poolAddresses: string[]
+): Promise<VotingResponse> => {
+  const response = await fetch(VOTING_API_URL, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify({
+      block,
+      address,
+      poolAddresses
+    })
+  });
+
+  const payload = await response.json();
+  return payload.data;
+};
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+) {
+  const blockTag =
+    typeof snapshot === 'number' ? snapshot : await provider.getBlockNumber();
+
+  const params = {
+    smartChefs: {
+      __args: {
+        where: {
+          startBlock_lte: blockTag,
+          endBlock_gte: blockTag
+        },
+        first: 1000,
+        orderBy: 'block',
+        orderDirection: 'desc'
+      },
+      id: true,
+      startBlock: true,
+      endBlock: true
+    }
+  };
+
+  const results = await subgraphRequest(SMART_CHEF_URL, params);
+
+  if (!results) {
+    return;
+  }
+
+  try {
+    const poolAddresses = results.smartChefs.map((pool) => pool.id);
+    const promises = addresses.map((address) => {
+      return fetchVotingPower(address, blockTag, poolAddresses);
+    }) as ReturnType<typeof fetchVotingPower>[];
+    const votingPowerResults = await Promise.all(promises);
```

Comment:
Hi @tomyumswap sending one request per address won't scale for a large number of votes, please find an alternate way to get all data for all addresses in a single request
https://github.com/snapshot-labs/snapshot-strategies#checklist-for-adding-a-new-strategy